src/connectivity-api/connectivity-qt/connectivityqt/connectivity.cpp

```diff
@@ -387,6 +390,34 @@ void Connectivity::setHotspotEnabled(bool enabled)
 
     // TODO Remove this when SDK switch widget isn't broken
     d->m_propertyCache->setProperty("HotspotEnabled", enabled);
+
+    // HACK: For switching MediaTek /dev/wmtWifi adapter mode between AP/STA
+    //       when hotspot enabled state is changed.
+    try
+    {
+        const char* wmtWifi_dev = "/dev/wmtWifi";
+
+        QFileInfo wmtWifi_fileinfo(wmtWifi_dev);
+        if (wmtWifi_fileinfo.exists())
+        {
+            QFile wmtWifi_file(wmtWifi_dev);
+            wmtWifi_file.open(QIODevice::WriteOnly);
```

Comment:
Open will return false on failure, please handle this also (qdebug is fine)

https://doc.qt.io/qt-5/qfile.html#open